cmd/notation/manifest.go

```diff
@@ -8,22 +8,26 @@ import (
 	"oras.land/oras-go/v2/registry"
 )
 
-func getManifestDescriptorFromContext(ctx context.Context, opts *SecureFlagOpts, ref string) (ocispec.Descriptor, error) {
+func getManifestDescriptorFromContext(ctx context.Context, opts *SecureFlagOpts, ref string) (ocispec.Descriptor, registry.Reference, error) {
 	if ref == "" {
-		return ocispec.Descriptor{}, errors.New("missing reference")
+		return ocispec.Descriptor{}, registry.Reference{}, errors.New("missing reference")
 	}
 
 	return getManifestDescriptorFromReference(ctx, opts, ref)
 }
 
-func getManifestDescriptorFromReference(ctx context.Context, opts *SecureFlagOpts, reference string) (ocispec.Descriptor, error) {
+func getManifestDescriptorFromReference(ctx context.Context, opts *SecureFlagOpts, reference string) (ocispec.Descriptor, registry.Reference, error) {
 	ref, err := registry.ParseReference(reference)
 	if err != nil {
-		return ocispec.Descriptor{}, err
+		return ocispec.Descriptor{}, registry.Reference{}, err
 	}
 	repo, err := getRepositoryClient(opts, ref)
 	if err != nil {
-		return ocispec.Descriptor{}, err
+		return ocispec.Descriptor{}, registry.Reference{}, err
 	}
-	return repo.Resolve(ctx, ref.ReferenceOrDefault())
+	manifestDesc, err := repo.Resolve(ctx, ref.ReferenceOrDefault())
```

Comment:
We might want to discourage the use of default `latest`. /cc @yizha1 @FeynmanZhou 
Note that `oras` CLI requires the customer to input the tag or the digest. Empty reference means errors.
Meanwhile, `docker` will use the default `latest` tag for empty reference.

@patrickzheng200 if we still want the default `latest`. We should do
```suggestion
	ref.Reference = ref.ReferenceOrDefault()
	manifestDesc, err := repo.Resolve(ctx, ref.Reference)
```

Reply:
I vote for "empty means errors, tag or digest is a MUST"

Reply:
Added check to return error if tag or digest is missing.

Reply:
@patrickzheng200 @shizhMSFT  Agree with avoiding using the default `lastest` if users don't specify a tag or digest. The registry side doesn’t tag anything automatically, so we need to prompt users explicitly on the client side.

Reply:
> @patrickzheng200 @shizhMSFT Agree with avoiding using the default `lastest` if users don't specify a tag or digest. The registry side doesn’t tag anything automatically, so we need to prompt users explicitly on the client side.

updated code to reflect this.